library/clinical_rules.php

```diff
@@ -2048,12 +2399,37 @@ function resolve_action_sql($rule, $group_id = '')
  * @param  string  $patient_id  pid of selected patient.
  * @param  array   $filter      array containing filter/target elements
  * @param  array   $interval    array containing interval elements
- * @param  string  $dateTarget  target date(format Y-m-d H:i:s). blank is current date.
+ * @param  string  $dateFocus   date for determining left boundary of interval (format Y-m-d H:i:s)
+ * @param  string  $dateTarget  date for determining right boundary of interval (format Y-m-d H:i:s). blank is current date.
  * @return boolean              true if check passed, otherwise false
  */
-function database_check($patient_id, $filter, $interval = '', $dateTarget = '')
+ /*
+   HR: is called for processing both filters and targets
+   When called for filters, $interval and $dateFocus are empty strings, and database_check() will look for data with start date prior to $dateTarget
+   When called for targets, $interval is valued (something like "1 year").
+   database_check() will look for data between $dateFocus - $interval and $dateTarget
+   (was previously looking for data between $dateFocus - $interval and $dateFocus. I changed to use $dateTarget instead for right interval boundary, for reasons mentioned below
+   in comments at sql_interval_string()
+   Targets are typically processed with $dateFocus set to: D, D+warningInterval or D-pastDueInterval, where D is timepoint passed to test_rules_clinic()
+   D is typically now()
+   Filters were previously processed with these same three timepoints, but I changed them to be processed only once, at D
+
+   $interval is something like: 1 year. Becomes $intervalType and $intervalValue
+   $dateTarget, $intervalType and $intervalValue are passed to exist_custom_item() and exist_database_item()
+   These functions call sql_interval_string(), passing $dateTarget, $intervalType and $intervalValue to sql_interval_string()
+   sql_interval_string() builds the sql query string that checks patient data against an interval
+   The string sql_interval_string() was previously building filters for patient data in date range:
+   $dateTarget - (interval mentioned in $intervalType and $intervalValue) -> $dateTarget (where $dateTarget is the value passed to sql_interval_string(),
+   which comes from $dateFocus in test_rules_clinic() )
+   I modified it to instead create an interval that is:
+   $dateFocus - (interval mentioned in $intervalType and $intervalValue) -> $dateTarget
+   (where $dateFocus and $dateTarget passed to sql_interval_string() are the same as $dateFocus and $dateTarget in test_rules_clinic() )
+   See comments in sql_interval_string() for reasons for this change
+ */
+function database_check($patient_id, $filter, $interval = '', $dateFocus = '', $dateTarget = '', $rule = '')
```

Comment:
is the new $rule parameter being used in this function?

Reply:
no, sorry. that was for my logging. Will remove